r/tools/nixlibs.R

```diff
@@ -473,17 +473,19 @@ build_libarrow <- function(src_dir, dst_dir) {
   env_vars <- env_vars_as_string(env_var_list)
 
   cat("**** arrow", ifelse(quietly, "", paste("with", env_vars)), "\n")
-  status <- suppressWarnings(system(
+
+  status <- suppressWarnings(system2(
     paste(env_vars, "inst/build_arrow_static.sh"),
-    ignore.stdout = quietly, ignore.stderr = quietly
+    stdout = ifelse(quietly, NULL, TRUE),
+    stderr = ifelse(quietly, NULL, TRUE)
```

Comment:
I'm not certain I follow the logic, but I think this is what you want

```suggestion
    # If running quietly, capture stdout/stderr so it doesn't print,
    # else print to the console
    stdout = ifelse(quietly, TRUE, ""),
    stderr = ifelse(quietly, TRUE, "")
```